### PR TITLE
growpart: Add btrfs support

### PR DESCRIPTION
**Pull request by cgwalters**
Fedora Workstation is using this for 33, it's easy to add.

### CODE REVIEW

**Comment by cgwalters** [comment]
Pairs with https://github.com/coreos/coreos-assembler/pull/1687

**Review by jlebon — APPROVED** [review]
Don't know btrfs, but looks sane!

File: overlay.d/05core/usr/lib/dracut/modules.d/40ignition-ostree/coreos-growpart [hunk 1]
```diff
@@ -64,9 +64,16 @@ if [[ "${src}" =~ "/dev/mapper" ]]; then
         exit 0
     fi
 else
-    # Handle traditional disk/partitions
-    majmin=$(findmnt -nvr -o MAJ:MIN "$path" | tail -n1)
-    devpath=$(realpath "/sys/dev/block/$majmin")
+    if test "${TYPE:-}" = "btrfs"; then
+        # Theoretically btrfs can have multiple devices, but when
+        # we start we will always have exactly one.
+        devpath=$(btrfs device usage /sysroot | grep /dev | cut -f 1 -d ,)
```
Comment (jamescassell) [inline overlay.d/05core/usr/lib/dracut/modules.d/40ignition-ostree/coreos-growpart:70]: I wonder whether btrfs supports JSON output, to make this more reliable...

**Comment by jlebon** [comment]
Can you rebase this one more time to retrigger CI?

**Pull request merged**